dproc/dproc.go

```diff
@@ -0,0 +1,166 @@
+	} else {
+		return DataprocRequestParams{}, fmt.Errorf("dataproc migration for source %s not supported", sourceProfile.Driver)
+	}
+
+	jarFileUris := []string{"file:///usr/lib/spark/external/spark-avro.jar",
+		"gs://dataproc-templates-binaries/latest/java/dataproc-templates.jar",
+		jdbcParams["jar"]}
+
+	args := []string{"--template",
+		"JDBCTOSPANNER",
+		"--templateProperty",
+		"project.id=" + spProject,
+		"--templateProperty",
+		"jdbctospanner.jdbc.url=" + jdbcParams["url"],
+		"--templateProperty",
+		"jdbctospanner.jdbc.driver.class.name=" + jdbcParams["driver"],
+		"--templateProperty",
+		"jdbctospanner.sql=" + jdbcParams["sql"],
+		"--templateProperty",
+		"jdbctospanner.output.instance=" + spInstance,
+		"--templateProperty",
+		"jdbctospanner.output.database=" + spDb,
+		"--templateProperty",
+		"jdbctospanner.output.table=" + srcTable,
+		"--templateProperty",
+		"jdbctospanner.output.primaryKey=" + primaryKeys,
+		"--templateProperty",
+		"jdbctospanner.output.saveMode=" + saveMode,
+		"--templateProperty",
+		"jdbctospanner.output.batch.size=" + batchSize,
+		"--templateProperty",
+		"jdbctospanner.jdbc.fetchsize=" + fetchSize}
+
+	dataprocRequestParams := DataprocRequestParams{
+		Project:       spProject,
+		TemplateArgs:  args,
+		JarFileUris:   jarFileUris,
+		SubnetworkUri: subnet,
+		Location:      location,
+	}
+
+	return dataprocRequestParams, nil
+}
+
+func CreateDataprocBatchClient(location string) (*dataproc.BatchControllerClient, error) {
+	ctx := context.Background()
+	batchEndpoint := fmt.Sprintf("%s-dataproc.googleapis.com:443", location)
+	batchClient, err := dataproc.NewBatchControllerClient(ctx, option.WithEndpoint(batchEndpoint))
+	return batchClient, err
+}
+
+func TriggerDataprocTemplate(batchClient *dataproc.BatchControllerClient, srcTable string, srcSchema string, primaryKeys string, dataprocRequestParams DataprocRequestParams) (string, error) {
+	ctx := context.Background()
+
+	fmt.Printf("Triggering Dataproc template for %s.%s\n", srcSchema, srcTable)
+	req := &dataprocpb.CreateBatchRequest{
+		Parent: "projects/" + dataprocRequestParams.Project + "/locations/" + dataprocRequestParams.Location,
+		Batch: &dataprocpb.Batch{
+			RuntimeConfig: &dataprocpb.RuntimeConfig{
+				Version: runtimeVersion,
+			},
+			EnvironmentConfig: &dataprocpb.EnvironmentConfig{
+				ExecutionConfig: &dataprocpb.ExecutionConfig{
+					Network: &dataprocpb.ExecutionConfig_SubnetworkUri{
+						SubnetworkUri: dataprocRequestParams.SubnetworkUri,
+					},
+				},
+			},
+			BatchConfig: &dataprocpb.Batch_SparkBatch{
+				SparkBatch: &dataprocpb.SparkBatch{
+					Driver: &dataprocpb.SparkBatch_MainClass{
+						MainClass: mainClass,
+					},
+					Args:        dataprocRequestParams.TemplateArgs,
+					JarFileUris: dataprocRequestParams.JarFileUris,
+				},
+			},
+		},
+	}
+
+	op, err := batchClient.CreateBatch(ctx, req)
+	if err != nil {
+		fmt.Printf("error creating the batch: %s\n", err.Error())
```

Comment:
please use the logger

Reply:
Done,
Commit: https://github.com/cloudspannerecosystem/harbourbridge/pull/520/commits/a245b849a66861ec60e6d44c3294e6e052f233bb